File: backend/app/services/cliente_service.py
import uuid

from app.domain.models import Cliente
from app.domain.schemas import ClienteCreate
from app.repositories.cliente_repo import ClienteRepository


class ClienteService:
    def __init__(self, repo: ClienteRepository):
        self._repo = repo

    async def criar(self, dados: ClienteCreate) -> Cliente:
        cliente = Cliente(nome=dados.nome, pasta_sharepoint_id=dados.pasta_sharepoint_id)
        return await self._repo.criar(cliente)

    async def obter(self, cliente_id: uuid.UUID) -> Cliente | None:
        return await self._repo.obter(cliente_id)

    async def listar(self) -> list[Cliente]:
        return await self._repo.listar()

File: frontend/src/app/core/api/services/cliente.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../../environments/environment';
import { ClienteCreate, ClienteRead } from '../models/cliente.model';

@Injectable({ providedIn: 'root' })
export class ClienteApiService {
  private readonly baseUrl = `${environment.apiUrl}/clientes`;

  constructor(private readonly http: HttpClient) {}

  listar(): Observable<ClienteRead[]> {
    return this.http.get<ClienteRead[]>(this.baseUrl);
  }

  obter(clienteId: string): Observable<ClienteRead> {
    return this.http.get<ClienteRead>(`${this.baseUrl}/${clienteId}`);
  }

  criar(dados: ClienteCreate): Observable<ClienteRead> {
    return this.http.post<ClienteRead>(this.baseUrl, dados);
  }

  atualizarDestinatarios(clienteId: string, destinatariosRelatorio: string): Observable<ClienteRead> {
    return this.http.patch<ClienteRead>(`${this.baseUrl}/${clienteId}/destinatarios`, {
      destinatarios_relatorio: destinatariosRelatorio,
    });
  }
}
